src/parser/ast.rs

```diff
@@ -933,7 +933,7 @@ pub struct Query {
 
 impl fmt::Display for Query {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{}{}", self.target, self.path)
+        write!(f, "{}.{}", self.target, self.path)
```

Comment:
Without this variable queries end up mushed together e.g. `foo.bar` becomes `foobar` however that then causes `.foo.bar` to become `..foo.bar` without the change to `QueryTarget` below.